overlord/devicestate/users.go

```diff
@@ -252,6 +252,7 @@ func getUserDetailsFromAssertion(assertDb asserts.RODatabase, modelAs *asserts.M
 	}
 	if len(su.Serials()) > 0 {
 		if serialAs == nil {
+			st.Set("system-user-assertion", "pending")
```

Comment:
this is kind of unexpected here. Can you remind why it was moved?

Reply:
The reason I moved it to the new place is that `[CreateKnownUsers](https://github.com/snapcore/snapd/blob/master/overlord/devicestate/users.go#L88)` (which calls  `getUserDetailsFromAssertion `) can be called from many different scenarios when system-user-assertion is processed, e.g. loaded from external storage, passed over REST API; not only as a result of first boot processing. In any of those scenarios `CreateKnownUsers` would fail if system-user-assertion is serial bound, and we do not have serial assertion yet. Moving place we set the state to "pending" to be closer to the actual error will cover all those various scenarios.

We can then handle "re-run" when we acquire serial assertion and if it is in a "pending" state, we can re-run `CreateKnownUsers`
For that reason I also renames the key, to be more related to the `system-user-assertion` rather than auto-import-assertion.